### DISCUSSION TITLE
Not a bug - request for better guide when changing parameters

### DISCUSSION THREAD

**Discussion by richardbelan** [klaasnicolaas/home-assistant-glow#425] [Ideas]
The purpose of this is not to report an issue. It's help to others, what they might consider, if the setup didn't work out as described.

**The photodiode/LDR**
I got similar photodiode print as described, but it was not detecting the pulses from the electricity meter. It was detecting the torch light/ambient light, also the red LED on the controller, but not the meter. My meter uses orange LED (R+G) and I have found out that the provided photodiode was apparently most sensitive to red/IR light.
So I have changed the photodiode to LDR with sensitivity around 540nm. That worked out of the box. So you can also consider to buy directly the light kit with LDR.
I highly recommend to put the LED/photodiode/LDR into black tubing (temperature shrink) kit (4.8 or 6.4mm) to avoid reaction to other light sources and cover the LEDs on the print.

**It is registering to many pulses**
Then it was working, but it was registering 3-4 times as much.
That was fixed by setting the filter to 200ms.
I found the documentation for how to achieve it, as unclear. 
So I have made my own version/compilation/build.

**How I made my own version/build**
It is my first time working with ESPHome before and I find the documentation both here and on ESPHome like black magic.
So I did it my way:
1. Installed everything as described
2. Installed espHome (developer) version on my Home Assistant
3. "Followed as much from ESPHome description as I could"
4. Forked this repo
5. change/add api: encryption key provided from (2) in the esp32.yaml. If you don't do that, you will miss connection to your Home Assistant when you upload the new firmware, but the server will still be running. If that happens, reinstall the original firmware:
api:
  encryption:
    key: 6......
6. change the internal filter option in pulse_meter.yaml
....
pin: ${pulse_pin}
internal_filter: 200ms

7. Downloaded the command line esphome compiler - it is via python/pip command on their webside.
8. went to the repo\home-assistant-glow, where the esp32.yaml (or similar) resides
9. ran: 
> esphome compile esp32.yaml
10. found the firmware somewhere if the compilation chain
11. uploaded the firmware.
If anybody has a better way how to do it, pls. comment it here. I was not able to find "esphome for dummies"

**Unsolved issues**
I am using esp32s and I find the power usage of this setup relatively high. So that is my next project to do something about it. The "mini" board seems to use less power, still this is factor 10 higher then what I would expect. Anybody has better solutions?
One thing could be to report the data less frequently and have some deep sleep. Others recommend to set unused pins into pull-up (or mount resistors).

**Comment by klaasnicolaas** [comment]
I have converted it into a discussion, everyone can make adjustments to the documentation, so this is also an open invitation to improve where it is unclear.

A side note, the step-by-step plan mentioned is an option that is actually unnecessarily complex. In the future it will only become easier to make adjustments (from version v4.0.1, not yet released).

./Klaas

**Reply by FionnD** [reply]
@klaasnicolaas 

You mentioned it's become easier since 4.0.1. Could you elaborate please?

**Reply by klaasnicolaas** [reply]
Nowadays you can use `!extend` to override settings like with the **internal_filter**
https://glow-energy.io/docs/advanced/firmware_changes

./Klaas

**Reply by FionnD** [reply]
Thank you. If I wish to switch the GPIO pin from 26 to 21 (I mistakenly attached it to the wrong pin), would the following be the correct syntax?

```
substitutions:
  - id: !extend subsititions_pulse_pin 
    pulse_pin: GPIO21
```

**Reply by klaasnicolaas** [reply]
I think that should work.
